.github/workflows/python-package-flake8.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: flake8
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  linting:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest pylint
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        # ngam, 2021-12-21: removed `--exit-zero` to make it fail
```

Comment:
@wkchuang essentially, this edit (taking out `--exit-zero` makes flake8 fail for the slightest error. Previously, it was lenient in that it would just report them as warnings, but not fail the build unless there is a dramatic error (line 34). We can try this for a while and see how if we want to customize it (there are two ways: 1) through a flake8 dotfile or 2) in this call right here).

Reply:
Cool, that's good to know, thanks!